### DISCUSSION TITLE
Why there is only one deep link allowed per screen?

### DISCUSSION THREAD

**Discussion by pke** [react-navigation/react-navigation#10384] [Q&A]
We can have the same Screen in more than one stack and can navigate to it using `navigate` which will always try to find the screen in the parent hierarchy. 

Deep links however are bound to one particular screen hierarchy only.

Wouldn't it be possible to be on Stack A or Stack B and a link "/details/:id" whould be mapped to a `Details` Screen in either of their stacks and depending on what stack the user currently is, the `Detail` is navigated too?

I think there is a reason why linking is only allowed to work the way it currently is, but I'd like to understand the reasoning behind it.

Could it be that if one is Stack C the system would not know how to show the deep link "/details" which would resolve to either Stack A/B, which are both not active currently?

```
<Tabs>
  <Screen Home>
    <Stack>
       <Screen Details>
  <Screen Search>
     <Stack>
       <Screen Details>
  <Screen Products>
    <Stack>
       <Screen Details>
   <Screen Profile>      
</Tabs>
```

```
{
  screens: {
    Home: {
      Details: "/details:id",
    },
    Search: {
      Details: "/details:id",
    },
    Products: {
      Details: "/details:id",
    }
  }
}
```

`myapp:/details/1` would open the Details screen in one of the 3 screens if one is currently open. But what if the "Profile" tab is open?

@satya164 any insights into the process of selecting the matching route?

**Comment by satya164** [comment]
> why linking is only allowed to work the way it currently is

you can do whatever you want if you customize how linking is handled using `getStateFromPath`. it's just complex to support such scenarios by default. when people open a deep link from a website, notification, etc., they unlikely have the context of what screen they were on previously anyway, so such behavior isn't really needed.

also, I don't think it really makes sense to show the details screen inside the search tab when coming from a deep link if the user didn't actually go there from search.

**Reply by pke** [reply]
Thanks for answering!

I am implementing getStateFromPath, yet double paths in Linking are complained about by the framework.

Where do you think details from Deep Links should be displayed then? Modal?

> when people open a deep link from a website, notification, etc., they unlikely have the context of what screen they were on previously anyway, so such behavior isn't really needed.

Thats the reason why I would want to put such details screen in more than one tab stack, so that users can easily navigate back from the deep link opened detail page
